src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbStreamsBySharedTable.java

```diff
@@ -79,9 +81,23 @@ protected MtGetRecordsResult getAllRecords(GetRecordsRequest request, StreamArn
             // when called to map a record, extract tenant context and table
             final FieldValue<?> fieldValue = fieldValueFunction.apply(record.getDynamodb().getKeys());
             // then establish context to map physical to virtual record, i.e., remove index key prefixes
-            return mtDynamoDb.getMtContext().withContext(fieldValue.getContext(), () ->
-                mtDynamoDb.getTableMapping(fieldValue.getTableName()).getRecordMapper().apply(record)
-            );
+            return mtDynamoDb.getMtContext().withContext(fieldValue.getContext(), () -> {
+                try {
+                    return mtDynamoDb.getTableMapping(fieldValue.getTableName()).getRecordMapper().apply(record);
+
+                } catch (UncheckedExecutionException e) {
+                    // This isn't great, but we're assuming a missing virtual table entry here is a deleted virtual
+                    // table, and thus, shouldn't be mapping records to MtRecords. Instead return null, but really,
+                    // we should make sure we let deleted records flow through a stream and expire out, before removing
+                    // the virtual table entry.
+                    if (e.getCause() instanceof ResourceNotFoundException) {
+                        return null;
```

Comment:
I wonder if returning and filtering out `null` could cause some downstream issues, because KCL now doesn't see these records, so it cannot checkpoint them. I wonder if instead we should just not convert keys or set dummy values or something.